app/Console/Commands/AddCoordinapeUserToAllCircles.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Profile;
+use App\Models\User;
+use App\Models\Circle;
+use Illuminate\Console\Command;
+
+class AddCoordinapeUserToAllCircles extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ape:populate_coordinate_user {address}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Command to add a Coordinate user to all circles with given address';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $address = strtolower($this->argument('address'));
+
+        $profile = Profile::firstOrCreate([
+            'address' => $address,
+        ]);
+
+        # Iterate across all circles (TODO: can we do this in one transaction?)
+        Circle::chunk(100, function ($circles) {
+            foreach ($circles as $circle) {
```

Comment:
@zashton silly question: what's the best way to pass $address into the context of the callback function here?

https://laravel.com/docs/8.x/collections#method-each

I'm getting: `  Undefined variable $address`

Reply:
nice, just added in for you. you can add a new user type for this as well if you want